src/main/scala/za/co/absa/abris/avro/sql/CatalystDataToAvro.scala

```diff
@@ -38,29 +40,30 @@ private[abris] case class CatalystDataToAvro(
   @transient private lazy val serializer: AvroSerializer =
     new AvroSerializer(child.dataType, config.schema, child.nullable)
 
+  @transient private lazy val writer =
+    new GenericDatumWriter[Any](config.schema)
+
+  @transient private var encoder: BinaryEncoder = _
+
+  @transient private lazy val out = new ByteArrayOutputStream
+
   override def nullSafeEval(input: Any): Any = {
-    val avroData = serializer.serialize(input)
+    out.reset()
 
-    val record : IndexedRecord = avroData match {
-      case ad: IndexedRecord => ad
-      case _ => wrapWithRecord(avroData)
+    config.schemaId.foreach { id =>
+      attachSchemaId(id, out)
     }
 
-    SparkAvroConversions.toByteArray(record, record.getSchema, config.schemaId)
-  }
-
-  private def wrapWithRecord(avroData:Any) = {
-    val record = new GenericData.Record(wrapSchemaIfNeeded(config.schema))
-    record.put(0, avroData)
-    record
+    encoder = EncoderFactory.get().directBinaryEncoder(out, encoder)
+    val avroData = serializer.serialize(input)
+    writer.write(avroData, encoder)
+    encoder.flush()
+    out.toByteArray
   }
 
-  private def wrapSchemaIfNeeded(schema: Schema) = {
```

Comment:
I am not sure what was causing the need for it in the first place. May suspicion is that the old code was sometime generating a bare schema. 

Bare schema is just a name of the type without any record. Such a schema is not possible to register in the schema registry. I think the new schema generation code should never return it, and the only place when it could occur is when someone provides it directly. To keep Abris consistent with spark avro I choose to remove this.